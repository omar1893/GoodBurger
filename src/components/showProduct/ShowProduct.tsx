import React from 'react'
import { ShowProductProp } from '../../models/ShowProductProp.model';
import ShowProductStyles from './ShowProduct.module.css'
import { PriceFormatter } from '../../utils/basic/priceFormatter';

const ShowProduct = ({product}: ShowProductProp) => {

  return (
    <div className={ShowProductStyles.ProductContainer}>
      <div>Product: <span>{product.name}</span></div>
      <div>Price: <span>${PriceFormatter(product.price)}</span></div>
    </div>
  );
}

export default ShowProduct