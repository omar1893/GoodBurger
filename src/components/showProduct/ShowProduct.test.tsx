import { render } from "@testing-library/react";
import ShowProduct from "./ShowProduct";

describe("ShowProduct component", () => {
  it("should match snapshot", () => {
    const product = {
      name: "Product 1",
      price: 9.99,
      type: "SANDWICH"
    };
    const { asFragment } = render(<ShowProduct product={product} />);
    expect(asFragment()).toMatchSnapshot();
  });
});