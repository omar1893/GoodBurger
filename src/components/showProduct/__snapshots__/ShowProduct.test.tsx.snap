// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowProduct component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="ProductContainer"
  >
    <div>
      Product: 
      <span>
        Product 1
      </span>
    </div>
    <div>
      Price: 
      <span>
        $$9.99
      </span>
    </div>
  </div>
</DocumentFragment>
`;
